js/clients/README.md

```diff
@@ -0,0 +1,185 @@
+const { BcoinClient } = require('@summa-tx/bitcoin-spv-js-clients');
+```
+
+ES6 module:
+
+```js
+import { BcoinClient } from '@summa-tx/bitcoin-spv-js-clients';
+```
+
+### Use
+
+The BcoinClient proof fetcher exports a single constructor function, `BcoinClient`, which accepts an options Object to configure the client. This constructor is an extension of bcoin's `NodeClient`. See [bcoin.io](https://bcoin.io/api-docs/#configuring-clients) for more information about how to configure the client.
+
+```js
+const options = {
+  network: 'testnet',
+  port: 18332,
+  apiKey: 'api-key'
+};
+
+const client = new BcoinClient(options);
+```
+
+### Methods
+
+#### client.getBlockHeader(block)
+   * `@param {Hash|Number} block`
+   * `@returns {Promise}`
+
+Retrieve a block header.
+
+```js
+const client = new BcoinClient(options);
+const blockHeader = await client.getBlockHeader(150151);
+
+// block header
+// {
+//   "hash": "6f1003edd05cad861395225415160b5236968cc223fe982796b6e959c9651d44",
+//   "version": 536870912,
+//   "prevBlock": "0c4ea5e675941eca1909275f21903cef755069a03c57c10f4d4dadcdd7146daf",
+//   "merkleRoot": "9a249c682aba07943c8a1f9bd774a15d71372fcd7f3f9ee99e8c7aa022ae6aa0",
+//   "time": 1571661863,
+//   "bits": 545259519,
+//   "nonce": 8,
+//   "height": 100,
+//   "chainwork": "00000000000000000000000000000000000000000000000000000000000000ca"
+// }
+```
+
+#### client.getProof(txid)
+   * `@param {String} txid` - txid in big endian
+   * `@returns {Object}` - proof object
+
+Get a proof for a transaction.
+
+```js
+const client = new BcoinClient(options);
+const txid =
+
+// proof object
+// {
+//   version: {String},
+//   vin: {String},
+//   vout: {String},
+//   locktime: {String},
+//   tx_id: {String},
+//   tx_id_le: {String},
+//   index: {Number},
+//   confirming_header: {Object},
+//   intermediate_nodes: {String}
+// }
+```
+
+#### client.getHeader(block)
+* `@param {Hash|Number} block` - block height or hash
+* `@returns {Object} header` - header object
+
+Gets header by height or hash.
+```js
+const client = new BcoinClient(options);
+const block = 12345;
+const header = client.getHeader(block);
+
+// {
+//   raw: {String},
+//   hash: {String},
+//   hash_le: {String},
+//   height: {Number},
+//   prevhash: {String},
+//   prevhash_le: {String}
+//   merkle_root: {String},
+//   merkle_root_le: {String}
+// }
+```
+
+#### client.getMerkleProof(txid, height)
+* `@param {String}` `txid`
+* `@param {Number} height`
+* `@returns {[][]String, Number}` - a merkle proof and the index of the leaf
+
+ Validate the merkle tree of a block and then compute a merkle proof of inclusion for the txid.
+
+ ```js
+const client = new BcoinClient(options);
+const height = 12345;
+const txid = '3a4324234'
+const header = client.getMerkleProof(txid, height);
+
+// merkle proof
+// [
+//   '8e2d404a039a7a3e1768b161aa23546aab0444b73905bdd3d68b3d6f1769e8c0...',
+//   4
+// ]
+```
+
+#### client.getHeaderChainByCount(height, count, enc)
```

Comment:
The concept of passing an encoding type is a little bit vestigial at this point, I originally had it because sometimes we wanted to pretty print and sometimes we wanted the raw array buffers. I think it may be a good idea to eliminate the `enc` variable to simplify and always return json (or uint8arrays if preferred) and then the ser package can be used to convert between the two.

Reply:
Yeah that makes sense. I need to finish up this refactor and push all the changes I have.

Reply:
Going to return `Uint8Array`s for all of these since the primary use case is for the verifier code, and we can use the `ser` package to go back and forth as needed.

Reply:
Jk. I think I'll save this for a different time. For now use the `ser` package on the other side.